rotkehlchen/chain/bitcoin/bch/__init__.py

```diff
@@ -68,33 +68,23 @@ def get_bitcoin_cash_addresses_balances(accounts: List[BTCAddress]) -> Dict[BTCA
 
 def _check_haskoin_for_transactions(
         accounts: List[BTCAddress],
-) -> Dict[BTCAddress, Tuple[bool, FVal]]:
+) -> Dict[BTCAddress, bool]:
     """Checks if the BCH addresses have at least a transaction.
     May raise RemoteError or KeyError or DeserializationError"""
-    have_transactions = {}
+    have_transactions: Dict[BTCAddress, bool] = {}
     params = '|'.join(accounts)
     bch_resp = request_get_dict(url=f'{HASKOIN_BASE_URL}/bch/blockchain/multiaddr?active={params}')
     for entry in bch_resp['addresses']:
         if entry['address'] in accounts:
-            balance = satoshis_to_btc(deserialize_fval(
```

Comment:
Question. I didn't check but I was almost sure that the balances were not needed for the derivation. We actually check the transactions instead of the balances https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/bitcoin/xpub.py#L89 So we are not removing the logic for the derivation but the calculation of balances during the derivation

Reply:
Which is needed as there is no other place where xpub address balances happen.

Xpubs != accounts

Also since you already go through all transactions, and balances at xpub addition and go through all necessary remote endpoint queries (which can be 100-400 for big DBs) ... then **YOU SHOULD KEEEP THAT INFORMATION** and not throw it away hoping it may be requeried afterwards.

Reply:
Okey understood. Thanks

Reply:
Got it